Improve server packaging

gcotelli opened the pull request:
Moved methods used exclusively in the server code to JRPC-Server package, in order to not have this methods if you load only the Client group.

juliendelplanque commented on src/JRPC-Server/JRPCHTTPServer.class.st:
```diff
@@ -36,11 +36,6 @@ JRPCHTTPServer >> initialize [
 ]
 
-{ #category : #accessing }
-JRPCHTTPServer >> port [
```
This method is yet never sent but maybe it should be kept no?

gcotelli replied:
If you think is useful we can keep it. Anyway there are other server options that can be exposed, like `#debugMode` and `#serverUrl`, probably I will need it in the future

gcotelli commented:
I've recovered the `port` method and added `debugMode` option because it's better for debugging in case of test failures.
I've also added some tests to cover the notification API

juliendelplanque commented:
Ok nice, waiting for CI to build the PR and I merge and create a new minor release.

gcotelli commented:
Finally travis finished running the builds. It's ready to merge now :)

juliendelplanque commented:
New release after integration of this PR. https://github.com/juliendelplanque/JRPC/releases/tag/v2.1.0
